protobuf-format.md

```diff
@@ -0,0 +1,227 @@
+# Protobuf Event Format for CloudEvents - Version 1.0
+
+## Abstract
+
+[Protocol Buffers][proto-home] is a mechanism for marshalling structured data,
+this document defines how CloudEvents are represented using [version 3][proto-3]
+of that specification.
+
+In this document the terms *Protocol Buffers*, *protobuf*, and *proto* are used
+interchangeably.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Attributes](#2-attributes)
+3. [Data](#3-data)
+4. [Transport](#4-transport)
+5. [Examples](#5-examples)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how the
+elements defined in the CloudEvents specification are are represented using
+a protobuf schema.
+
+The [Attributes](#2-attributes) section describes the naming conventions and
+data type mappings for CloudEvents attributes for use as protobuf message
+properties.
+
+The [Data](#3-data) section describes how the event payload is carried.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Content-Type
+
+There is no official IANA *media-type* designation for protobuf, as such this
+specification uses 'application/protobuf' to identify such content.
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are represented in the protobuf
+[schema][proto-schema].
+
+## 2.1 Type System
+
+The CloudEvents type system is mapped to protobuf as follows :
+
+| CloudEvents   | protobuf |
+| ------------- | ---------------------------------------------------------------------- |
+| Boolean       | [boolean][proto-scalars] |
+| Integer       | [sint32][proto-scalars] |
```

Comment:
`sint32` is very rarely used within Google; we normally stick to `int32`. While `sint32` is more compact for negative numbers, it's (potentially) larger for non-negative numbers.

Reply:
(I note that it *is* `int32` in the spec.proto file.)

Reply:
(I suggested the switch to `int32`, but I think it only got corrected in the `.proto` file. It seems like we should make sure that changes in the `.proto` get reflected back into the text document.)

Reply:
Ack